### DISCUSSION TITLE
How to configure next-intl to support urls like /tenant-slug/locale/some-other-slug

### DISCUSSION THREAD

**Discussion by victor-mula** [amannn/next-intl#776] [Q&A] [answered]
I was wondering if it was possible at the moment a URL structure like  _/tenant-slug/locale/login_ where the locale is not at the beginning of the url. So when a user tries to access _/tenant-slug/login_ will get redirected to _/tenant-slug/en/login_. 

I haven't found any example in the docs or any open discussion about it.

Thanks in advance.

**Comment by amannn** [comment]
You're probably looking for this: https://github.com/amannn/next-intl/issues/653

**Reply by victor-mula** [reply]
Thanks for your quick answer.

I took a look at the solutions and tried to applied them mentally to my use case, and I have a hard time understanding the general strategy creating a custom middleware with rewrites.

So if I access a URL like: /tenant-slug/en/route
The idea is to extract the tenant-slug and then call the intl middleware? Does that make any sense?

```
const tenantSlug = request.nextUrl.pathname.split('/')[1];
const remainingPath = request.nextUrl.pathname.replace(`/${tenantSlug}`, '');
const newUrl = new URL(`${remainingPath}`, request.url);
const modifiedRequest = new NextRequest(newUrl, request);
  
const response = createIntlMiddleware({
    locales: ['en', 'de'],
    defaultLocale: 'en',
  })(modifiedRequest);

return response;
  ```
  
In my case i have a page like "[tenantSlug]/[locale]/login" where I need the tenantSlug information to conditionally render some content. So I cant simply redirect to the url without the tenant info

**Reply by amannn** [reply, marked as answer]
If you need to read the `tenantSlug` in one of your pages, then it needs to be in the final URL that is returned from the middleware. For the time being, you might be better off providing your own middleware and navigation APIs as it's described in https://github.com/amannn/next-intl/issues/609. You can still use the other APIs from `next-intl` like `useTranslations`.

I hope to provide better support in this area in the future but haven't had the time yet to dive deeper into this.
